lib/srv/db/snowflake/http.go

```diff
@@ -107,7 +108,7 @@ func maybeReadGzip(headers *http.Header, body []byte) ([]byte, error) {
 	}
 	defer bodyGZ.Close()
 
-	body, err = io.ReadAll(bodyGZ)
+	body, err = utils.ReadAtMost(bodyGZ, teleport.MaxHTTPRequestSize)
```

Comment:
We've already read at most 10MB of (potentially compressed) data on line 84, and now we're requiring that the uncompressed data is also below the same threshold. Is this what we want?

You can probably just remove this particular change. Our `gzip.NewReader` already enforces a 10MB limit, so this code is functionally equivalent to the prior version.

Reply:
That is the intention here.  The issue is that 10MB of compressed data can decompress to gigabytes of data which could easily impact the service.